nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -532,6 +532,34 @@ let
             '';
           });
 
+        createGreDevice = n: v: nameValuePair "${n}-netdev"
+          (let
+            deps = deviceDependency v.dev;
+            greType = if v.type == "tun" then "gre" else "gretap";
```

Comment:
this expression is duplicated a few times. you could add it to the option definition to have the option checked against the declared type, but stored for later use as though this expression had always been applied to it first.
```nix
  apply = v: if v == "tun" then "gre" else "gretap";
```

Reply:
Nice!  I didn't know about this option.